src/openassetio-core/tests/hostApi/ManagerTest.cpp

```diff
@@ -413,3 +413,346 @@ TEMPLATE_TEST_CASE(
+                              ErrorCode::kEntityResolutionError>)) {
+  namespace managerApi = openassetio::managerApi;
+  namespace hostApi = openassetio::hostApi;
+  using trompeloeil::_;
+
+  using ExpectedExceptionType = typename TestType::ExceptionType;
+  static constexpr ErrorCode kExpectedErrorCode = TestType::kErrorCode;
+
+  GIVEN("a configured Manager instance") {
+    const std::shared_ptr<managerApi::ManagerInterface> managerInterface =
+        std::make_shared<openassetio::MockManagerInterface>();
+
+    auto mockManagerInterfacePtr =
+        std::static_pointer_cast<openassetio::MockManagerInterface>(managerInterface);
+
+    // Create a HostSession with our mock HostInterface
+    const managerApi::HostSessionPtr hostSessionPtr = managerApi::HostSession::make(
+        managerApi::Host::make(std::make_shared<openassetio::MockHostInterface>()),
+        std::make_shared<openassetio::MockLoggerInterface>());
+
+    // Create the Manager under test.
+    const hostApi::ManagerPtr manager =
+        hostApi::Manager::make(mockManagerInterfacePtr, hostSessionPtr);
+
+    const openassetio::trait::TraitSet traits = {"fakeTrait", "secondFakeTrait"};
+    auto context = openassetio::Context::make();
+
+    AND_GIVEN(
+        "manager plugin will encounter an entity-specific error when next preflighting a "
+        "reference") {
+      const openassetio::EntityReference ref = openassetio::EntityReference{"testReference"};
+      const openassetio::EntityReferences refs = {ref};
+
+      const openassetio::BatchElementError expectedError{kExpectedErrorCode, "Some error message"};
+
+      // With success callback side effect
+      REQUIRE_CALL(*mockManagerInterfacePtr,
+                   preflight(refs, traits, context, hostSessionPtr, _, _))
+          .LR_SIDE_EFFECT(_6(123, expectedError));
+
+      WHEN("preflight is called with kException errorPolicyTag") {
+        THEN("an exception is thrown") {
+          try {
+            manager->preflight(ref, traits, context,
+                               hostApi::Manager::BatchElementErrorPolicyTag::kException);
+            FAIL_CHECK("Exception not thrown");
+          } catch (const ExpectedExceptionType& exc) {
+            CHECK(exc.what() == expectedError.message);
+            CHECK(exc.error == expectedError);
+            CHECK(exc.index == 123);
+          }
+        }
+      }
+    }
+
+    AND_GIVEN(
+        "manager plugin will encounter entity-specific errors when next preflighting multiple "
+        "references") {
+      const openassetio::EntityReferences refs = {openassetio::EntityReference{"testReference1"},
+                                                  openassetio::EntityReference{"testReference2"}};
+
+      const openassetio::BatchElementError expectedError{kExpectedErrorCode, "Some error"};
+
+      // With success callback side effect
+      REQUIRE_CALL(*mockManagerInterfacePtr,
```

Comment:
[fix] Same comments from https://github.com/OpenAssetIO/OpenAssetIO/pull/879#discussion_r1158746917, and likely the same places, as i took the base of that code from this PR :P

Reply:
Fixed in 0419a37c